scalability_and_performance/cnv-performance-scale-resilience-practices/recommended-node-practices.adoc

```diff
@@ -0,0 +1,86 @@
+
+```
+apiVersion: remediation.medik8s.io/v1alpha1
+kind: NodeHealthCheck
+metadata:
+  name: nodehealthcheck-sample
+spec:
+  minHealthy: <set to 1 for critical environments> 
+  pauseRequests: 
+    - <pause-test-cluster>
+  remediationTemplate: 
+    apiVersion: self-node-remediation.medik8s.io/v1alpha1
+    name: self-node-remediation-automatic-strategy-template
+    namespace: openshift-workload-availability
+    kind: SelfNodeRemediationTemplate
+  escalatingRemediations: 
+    - remediationTemplate:
+        apiVersion: self-node-remediation.medik8s.io/v1alpha1
+        name: self-node-remediation-resource-deletion-template
+        namespace: openshift-workload-availability
+        kind: SelfNodeRemediationTemplate
+    order: 1
+    timeout: 300s
+  selector: 
+    matchExpressions:
+      - key: node-role.kubernetes.io/worker
+        operator: Exists
+  unhealthyConditions: 
+    - type: Ready
+      status: "False"
+      duration: <set to lower timings for faster recovery, for example 20s> 
+    - type: Ready
+      status: Unknown
+      duration: <set to lower timings for faster recovery, for example 20s>
+```
+
+Node Health Check operator supports Self Node Remediation (SNR) or Fence Agents Remediation (FAR) as the remediation operator to apply after detecting a node failure.
+
+- Self Node Remediation (SNR): Recommended for nodes that do not have a management interface, or the interface might be unreachable, since SNR does not require one to work.
+
+
+- Fence Agents Remediation (FAR): Fastest remediation operator in terms of workload recovery time. It also requires a management interface. FAR is recommended when a management interface is available. Also, the management interface must be reachable from the Kubernetes pod network to work.
+
+For more information on remediation, fencing, and maintaining nodes, see the link:https://access.redhat.com/documentation/en-us/workload_availability_for_red_hat_openshift/23.2/html-single/remediation_fencing_and_maintenance/index#about-remediation-fencing-maintenance[Workload Availability for Red Hat OpenShift] documentation.
+
+
+- Capacity planning to ensure nodes have enough resources to host VMs migrated from the failed node(s). It is recommended to monitor the VMs count, resource available -  CPU, memory, disk, network on the cluster and ensure that there are enough resources on the nodes to host the VMs migrated from the failed nodes.
+
+
+[IMPORTANT]
+====
+- The remediation actions starts after the NHC duration and can take large amounts of time depending on the number of VMS on the failed nodes. For example, 79 VMs with the following spec took 750 seconds to recover with SNR and will be much faster with FAR remediation strategy.
+
+- FAR is recommended if the environment has access to baseboard management controller (BMC) interface as the it remediates and recovers the VMs quicker when compared to SNR.
```

Comment:
```suggestion
- FAR is recommended if the environment has access to baseboard management controller (BMC) interface because it is able to remediate and recover VMs quicker than SNR.
```